pcstac/pcstac/filter.py

```diff
@@ -0,0 +1,102 @@
+        When queryables are requested without specifying a collection (/queryable
+        from the root), a json schema encapsulating only the properties shared by all
+        collections should be returned. This function gathers all collection
+        queryables, calculates the intersection, and caches the results so that the
+        work can be saved for future queries.
+        """
+        pool = request.app.state.readpool
+
+        async with pool.acquire() as conn:
+            collections = await conn.fetchval(
+                """
+                SELECT * FROM all_collections();
+                """
+            )
+        collection_ids = [collection["id"] for collection in collections]
+        all_queryables = await asyncio.gather(
+            *[self.get_queryables(cid) for cid in collection_ids],
+            return_exceptions=True
+        )
+        all_queryables = [
+            queryable
+            for queryable in all_queryables
+            if not isinstance(queryable, Exception)
+        ]
+        all_properties: List[dict] = [
+            queryable["properties"] for queryable in all_queryables
+        ]
+        all_property_keys: List[Set[str]] = list(
+            map(lambda x: set(x.keys()), all_properties)
+        )
+        intersecting_props = {}
+        if len(all_property_keys) > 0:
+            property_name_intersection: List[str] = list(
+                set.intersection(*all_property_keys)
+            )
+            for name in property_name_intersection:
+                for idx, properties in enumerate(all_properties):
+                    if idx == 0:
+                        maybe_match = properties[name]
+                    else:
+                        if properties[name] != maybe_match:
+                            break
+                if maybe_match is not None:
+                    intersecting_props[name] = maybe_match
+
+        intersection_schema = {
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "$id": "https://example.org/queryables",
+            "type": "object",
+            "title": "",
+            "properties": intersecting_props,
+        }
+        return intersection_schema
+
+    async def get_queryables(
+        self, collection_id: Optional[str] = None, **kwargs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Get the queryables available for the given collection_id.
+        If collection_id is None, returns the intersection of all
+        queryables over all collections.
+        This base implementation returns a blank queryable schema. This is not allowed
+        under OGC CQL but it is allowed by the STAC API Filter Extension
+        https://github.com/radiantearth/stac-api-spec/tree/master/fragments/filter#queryables
+        """
+        if not collection_id:
+            try:
+                queryable_resp = queryables_endpoint_cache["/collections"]
```

Comment:
Looks like the wrong key for the cache.

```suggestion
                queryable_resp = queryables_endpoint_cache["/queryables"]
```

Reply:
Woof. Nice catch!